### PR TITLE
Convert Language enum to struct

### PR DESCRIPTION
**Pull request by krzyzanowskim**
Convert `Language` to a struct.

Reasoning:
- It is defined as a string parameter of `TextDocumentItem` (specification recommends a lot more identifiers, that `sourcekit-lsp` won't support)
- as a struct, it can be extended outside the `LanguageServerProtocol` module if needed.

### CODE REVIEW

File: Sources/LanguageServerProtocol/Language.swift [hunk 1]
```diff
@@ -11,23 +11,18 @@
...
-  case c
-  case cpp // C++, not C preprocessor
-  case objective_c = "objective-c"
-  case objective_cpp = "objective-cpp"
-  case swift
+public struct Language: RawRepresentable, Codable, Equatable, Hashable {
+  public typealias LanguageId = String
 
-  case unknown
+  public static let c = Language(rawValue: "c")
```
Comment (benlangmuir) [inline Sources/LanguageServerProtocol/Language.swift]: Please add type annotations to all struct/class properties.

File: Sources/LanguageServerProtocol/Language.swift [hunk 2]
```diff
@@ -11,23 +11,18 @@
...
 
 /// A source code language identifier, such as "swift", or "objective-c".
-public enum Language: String, Codable, Hashable {
-  case c
-  case cpp // C++, not C preprocessor
-  case objective_c = "objective-c"
-  case objective_cpp = "objective-cpp"
-  case swift
+public struct Language: RawRepresentable, Codable, Equatable, Hashable {
+  public typealias LanguageId = String
```
Comment (benlangmuir) [inline Sources/LanguageServerProtocol/Language.swift]: Is it useful to have the extra type `Language.LanguageId`, or should we just use `RawValue` which is needed anyway?

File: Sources/LanguageServerProtocol/Language.swift [hunk 3]
```diff
@@ -38,12 +33,22 @@ public enum Language: String, Codable, Hashable {
...
+      default: return nil
     }
   }
 
   /// Clang-compatible language name for a header file. See `xflag`.
   public var xflagHeader: String? {
     return xflag.map { "\($0)-header" }
   }
+
+  public static func ~= (lhs: Language, rhs: Language) -> Bool {
```
Comment (benlangmuir) [inline Sources/LanguageServerProtocol/Language.swift]: Why is this needed? I would have expected being Equatable was enough.

Comment (krzyzanowskim) [reply Sources/LanguageServerProtocol/Language.swift]: not needed

File: Sources/LanguageServerProtocol/Language.swift (hunk 2 above)
Comment (krzyzanowskim) [reply Sources/LanguageServerProtocol/Language.swift]: I like to cover generic types with more descriptive alias, but I can ditch it if you prefer not

Comment (benlangmuir) [reply Sources/LanguageServerProtocol/Language.swift]: Okay, doesn't make much difference to me.

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
```
CodingTests.swift:206: error: CodingTests.testValueCoding : XCTAssertEqual failed: ("unknown") is not equal to ("asdfAF") - 
```

File: Sources/SKTestSupport/UnknownLanguage.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+
+public extension Language {
+  static let unknown: Language = Language(rawValue: "unknown")
```
Comment (benlangmuir) [inline Sources/SKTestSupport/UnknownLanguage.swift]: Is this used anywhere other than the line
```
checkCoding(Language.unknown, json: "\"unknown\"")
```

in the CodingTests.swift above the one you removed?  If so, we don't need to expose this static property at all and can just check coding on `Language(rawValue: "unknown")`.

Comment (krzyzanowskim) [reply Sources/SKTestSupport/UnknownLanguage.swift]: done

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test

**Comment by benlangmuir** [comment]
@swift-ci please test
